cclib/parser/xtbparser.py

```diff
@@ -0,0 +1,491 @@
+    def __init__(self, *args, **kwargs):
+        # Call the __init__ method of the superclass
+        super(XTB, self).__init__(logname="XTB", *args, **kwargs)
+
+    def __str__(self):
+        """Return a string representation of the object."""
+        return "XTB log file %s" % (self.filename)
+
+    def __repr__(self):
+        """Return a representation of the object."""
+        return 'XTB("%s")' % (self.filename)
+
+    def extract(self, inputfile, line):
+        # Extract xtb version
+        if '* xtb version' == line.strip()[:13]:
+            version = line.split()[3]
+            self.metadata["legacy_package_version"] = version
+            self.set_attribute('xtbversion', version)
+        
+        #   -------------------------------------------------
+        #  |                Calculation Setup                |
+        #    -------------------------------------------------
+        #
+        #   program call               : <command>
+        #   hostname                   : <hostname>
+        #   coordinate file            : <filename>.<<filetype>
+        #   omp threads                :                    12
+        #   number of atoms            :                    18
+        #   number of electrons        :                    66
+        #   charge                     :                     0
+        #   spin                       :                   0.0
+        #   first test random number   :      0.87181443679343
+
+        if 'coordinate file' == line.strip()[:15]:
+            self.metadata["coord_type"] = line.split()[3].split('.')[-1].lower()
+        
+        # Grab total charge
+        if 'charge' == line.strip()[:6]:
+            charge = int(line.split()[2])
+            self.set_attribute('charge', charge)
+        
+        # Multiplicity = Spin + 1
+        if 'spin' == line.strip()[:4]:
+            spin = float(line.split()[2]) + 1
+            self.set_attribute('mult', spin)
+
+        # Grabbing SCF energies from geometry optimization steps
+        #
+        # ........................................................................
+        # .............................. CYCLE    1 ..............................
+        # ........................................................................
+        #    1    -37.9590419 -0.379590E+02  0.116E-04    2.97       0.0  T
+        #    2    -37.9590419 -0.674305E-11  0.982E-05    2.97     240.0  T
+        #    3    -37.9590419 -0.129340E-09  0.425E-05    2.97     554.7  T
+        #      SCC iter.                  ...        0 min,  0.004 sec
+        #      gradient                   ...        0 min,  0.002 sec
+        #  * total energy  :   -37.3373741 Eh     change       -0.1278963E-08 Eh
+        #    gradient norm :     0.0004128 Eh/α   predicted     0.0000000E+00 (-100.00%)
+        #    displ. norm   :     0.0058476 α      lambda       -0.2852929E-06
+        #    maximum displ.:     0.0054052 α      in ANC's #1, #7, #3, ...
```

Comment:
This output is dependent on the optimizer used, `xtb` currently supports two variants of ANCopt and the RF based one is exchanged for the LBFGS based at 500 atoms. Also, the user can configure this in the input file always use the RF or LBFGS based version.